fuel-core/src/service/modules.rs

```diff
@@ -109,20 +137,37 @@ pub async fn start_modules(config: &Config, database: &Database) -> Result<Modul
 
     block_importer.start().await;
 
-    bft.start(
-        relayer_sender.clone(),
-        tx_request_event.clone(),
-        block_producer.clone(),
-        block_importer.sender().clone(),
-        block_importer.subscribe(),
-    )
-    .await;
+    match &coordinator {
+        CoordinatorService::Poa(poa) => {
+            // TODO: this must be connected to the block import mechanism
+            let (block_import_tx, _block_import_rx) = broadcast::channel(16);
+
+            poa.start(
+                txpool_builder.subscribe(),
+                txpool_builder.sender().clone(),
+                block_import_tx,
+                block_producer.clone(),
+                database.clone(),
+            )
+            .await;
+        }
+        CoordinatorService::Bft(bft) => {
+            bft.start(
+                relayer_sender.clone(),
+                tx_request_event.clone(),
+                block_producer.clone(),
+                block_importer.sender().clone(),
+                block_importer.subscribe(),
+            )
+            .await;
+        }
+    }
 
     sync.start(
         rx_block,
         tx_request_event.clone(),
         relayer_sender,
-        bft.sender().clone(),
+        // bft.sender().clone(),
```

Comment:
Is there a more clear way we can distinguish BFT is disabled because node is currently in PoA configuration

Reply:
Just include in the commeent as to why `bft.sender().clone()` is commented out rather than not outright deleted

Reply:
Sure